Add Windows Guide (#1380)
privacyguides/privacyguides.org: docs/windows/hardening.md

```

![TPM + PIN](/assets/img/windows/TPM%20%2B%20PIN%20Bitlocker.webp)

!!! tip
    To go to Computer Configuration, Just search Group Policy in the Windows Search Bar and Enter. You will see it and proceed as mentioned above.

When you do this, the encryption keys of your drive are only unlocked once you enter the PIN, and the decryption happens after. If you forget or lose the PIN, you won't be able to access your drives and OS anymore, and the only way to recover is using the Recovery Key provided during the initial setup of BitLocker. **Make sure you store it in a safe place**, such as a password manager, and keep backups of your Recovery Key.

<!-- ## Privacy

In order to limit data harvesting from Microsoft, several things can be done.

You must follow this [guide](https://docs.microsoft.com/en-us/windows/privacy/manage-connections-from-windows-operating-system-components-to-microsoft-services) first, it can take about 1 hour to get these things done.--> 

## Security Hardening :

### **Creation of User Account and usage**

> - Create another `administrator` account and change your current account to a `standard` user account to reduce the attack surface enormously as Most vulnerabilities today come from the fact that the user is always in `administrator` mode.

Reviewer: ```suggestion
- Create another `administrator` account and change your current account to a `standard` user account to reduce the attack surface enormously as most vulnerabilities today come from the fact that the user is always in `administrator` mode.
```

- Don't use Admin account for any of your personal tasks ! 

- Just restrict it to the Standard account created.

- Set UAC Settings to maximum by just search for User Account Control in Windows Search, Enter set the slider to the top.
    

> - Only use your account for you, if someone needs to use your computer **ALWAYS** create another standard account, even if it's for a one-time use, even if it's your family or someone you trust. This person can plug an malicious USB, can connect to malicious Wi-Fi Netowrk, download infectious file, etc... without you knowing about it.

Reviewer: ```suggestion
- Only use your account for you, if someone needs to use your computer **ALWAYS** create another standard account, even if it's for a one-time use, even if it's your family or someone you trust. This person can plug a malicious USB, can connect to malicious Wi-Fi network, download infectious file, etc... without you knowing about it.
```

!!! tip

>     You should ALWAYS do the quick shortcut ++win+l++ to lock your device when your away to prevent Unautorized access.

Reviewer: ```suggestion
    You should ALWAYS do the quick shortcut ++win+l++ to lock your device when you are away to prevent unauthorized access.
```

### **Protection against Malware and Viruses**

- Just use the Built-in Windows Defender Security to protect against threats.
    
    
!!! tip

>         Disable automatic Sample Submission in Windows defender Settings as Windows defender will send your files as a samples for Signature Database and might leak your data.

Reviewer: ```suggestion
        Disable automatic Sample Submission in Windows defender Settings as Windows defender will send your files as a samples for Signature Database and might leak your data.
        **Enable** the Group Policy `Computer Configuration > Administrative Templates > Windows Components > Microsoft Defender Antivirus > MAPS > Send file samples when further analysis is required to Never Send.`
```

![user-account](/assets/img/windows/user-account.webp)

!!! note
    Microsoft is pushing users to use  Microsoft accounts for other editions except Education and Enterprise after installation. 

    So, You could follow the guide by [ghacks.net](https://www.ghacks.net/2022/05/13/how-to-bypass-the-microsoft-account-requirement-during-windows-setup/) to bypass the Microsoft account requirement during setup and use Local account.

While setting up, it is recommended to use a generic names such as `user` and `host` and avoid identifying terms such as your name or operating system. This can make it more difficult for privileged `Win32` apps or attackers to discern your identity.

![Privacy Settings](/assets/img/windows/privacy-settings.webp)
<!-- Need to add more images regarding this here which will be provided by d4rklynk or list them using Admonitions--->
<!--Toggle off all privacy related settings as shown in the image :

- [ ]-->

## Encrypting the Drive

After you have installed Windows, turn on Full Disk Encryption (FDE) using BitLocker via Control Panel. 

!!! info "Choosing the Way to Encrypt"

>     It is recommended to use only the Control Panel because if you go to encrypt via Settings, Microsoft named it as `Device Encryption` and designed it in a way that the encryption keys for BitLocker would be stored on Microsoft's server which is attached to your Micrsoft account. This can be dangerous to your privacy and security as Microsoft could easily view your encrypted files, as could an attacker if they were able to gain access to Microsoft's servers or any Law Enforcement could by a Gag order. 

Reviewer: ```suggestion
    It is recommended to use only the Control Panel because if you go to encrypt via the settings app, Microsoft named it as `Device Encryption` and designed it in a way that the encryption keys for BitLocker would be stored on Microsoft's server which is attached to your Micrsoft account. This can be dangerous to your privacy and security as Microsoft could easily view your encrypted files, as could an attacker if they were able to gain access to Microsoft's servers or any Law Enforcement could by a Gag order. 
```

![Bitlocker in Control Panel](/assets/img/windows/bitlocker-control%20panel.webp)

As you are using Windows 11, TPM is used to encrypt and decrypt the drive but it is susceptible to [cold boot attacks](https://blog.elcomsoft.com/2021/01/understanding-bitlocker-tpm-protection/). So, it is recommended to use TPM + PIN by configuring the Group Policy Editor by going to the location below and selecting things as shown in the image below it: 

```
Computer Configuration > Administrative Templates > Windows Components > Bitlocker Drive Encryption > Operating System Drives > Require Additional Authentication at startup
```

![TPM + PIN](/assets/img/windows/TPM%20%2B%20PIN%20Bitlocker.webp)

Also enable this Group Policy

```
Computer Configuration > Administrative Templates > Windows Components > Bitlocker Drive Encryption > Operating System Drives > Allow Enchanced PIN for startup

```
![Enhanced Pins](/assets/img/windows/enhanced-pins.webp)

This will allow to you use Alphanumeric characters for the Pre-boot PIN.

> 

Reviewer: ```suggestion
After enabling these policies, you need to create your PIN by opening **Windows Terminal** as an **administrator**.
You can copy paste `manage-bde -protectors -add c: -TPMAndPIN` and it will prompt you to enter a new PIN.
You can check if it's enabled by typing `manage-bde -status`. It will normally show in **Key Protectors** : Numerical Password (it's the recovery key) and TPM And PIN.

!!! info
If you want to disable pre-boot PIN requirements. Set the 2 policies we did to **Not configured**, open **Windows Terminal** as an administrator and copy/paste this command `manage-bde -protectors -add c: -TPM`. You can again check if it worked by typing `manage-bde -status c:` and it will show you **Numerical Password** and **TPM**
```

<!-- ## Privacy

In order to limit data harvesting from Microsoft, several things can be done.

You must follow this [guide](https://docs.microsoft.com/en-us/windows/privacy/manage-connections-from-windows-operating-system-components-to-microsoft-services) first, it can take about 1 hour to get these things done.
it can take about 1 hour to get these things done. You can skip some of these chapters, like : 
- **7**
- **8** if you use Internet Explorer
- **11** if you use Mail
- **12** if you Microsoft Account (like Office 365 or OneDrive)
- **13** if you use Edge
- **16** if you use OneDrive
--> 

## Security Hardening

### **Creation of User Account and usage**

- Create another `administrator` account and change your current account to a `standard` user account to reduce the attack surface enormously as most vulnerabilities today come from the fact that the user is always in `administrator` mode.

- Don't use Admin account for any of your personal tasks ! 

- Just restrict it to the Standard account created.

> - Set [UAC](https://docs.microsoft.com/en-us/windows/security/identity-protection/user-account-control/user-account-control-security-policy-settings) Settings to the [Highest Privelege](https://support.microsoft.com/en-us/windows/about-user-account-control-settings-d5b2046b-dcb8-54eb-f732-059f321afe18).

Reviewer: ```suggestion
- Set [UAC](https://docs.microsoft.com/en-us/windows/security/identity-protection/user-account-control/user-account-control-security-policy-settings) settings to the [Highest Privelege](https://support.microsoft.com/en-us/windows/about-user-account-control-settings-d5b2046b-dcb8-54eb-f732-059f321afe18).
```

### **Protection against Malware and Viruses**

- Just use the Built-in Windows Defender Security to protect against threats.
    
    
!!! tip
    - Disable automatic Sample Submission in Windows defender Settings as Windows defender will send your files as a samples for Signature Database and might leak your data.
    - **Enable** the Group Policy `Computer Configuration > Administrative Templates > Windows Components > Microsoft Defender Antivirus > MAPS > Send file samples when further analysis is required to Never Send.`

- Enable [Microsoft Defender Application Guard](https://docs.microsoft.com/en-us/windows/security threat-protection/microsoft-defender-application-guard/md-app-guard-overview). After installing by going to "[Turn Windows Features on or off](https://docs.microsoft.com/en-us/windows/security/threat-protection/microsoft-defender-application-guard/install-md-app-guard)" you can enable it. This runs Microsoft Edge and other **Microsoft** Apps such as MS office,etc in an Isolated Hyper-V container preventing unknown Malware from damaging the system.

> - Enable [Memory Integrity](https://support.microsoft.com/en-us/windows/core-isolation-e30ed737-17d8-42f3-a2a9-87521df09b78) (also called Hypervisor-Protected Code Integrity) in Windows Defender Settings.

Reviewer: ```suggestion
- Enable [Memory Integrity](https://support.microsoft.com/en-us/windows/core-isolation-e30ed737-17d8-42f3-a2a9-87521df09b78) (also called Hypervisor-Protected Code Integrity) in Windows Defender settings.
```

<!-- - Enable [Secure Launch](https://docs.microsoft.com/en-us/windows/security/threat-protection/windows-defender-system-guard/how-hardware-based-root-of-trust-helps-protect-windows). You can [enable](https://docs.microsoft.com/en-us/windows/security/threat-protection/windows-defender-system-guard/system-guard-secure-launch-and-smm-protection) it by following the instructions.-->

- Enable `Display File Extensions` as most problems start here
    
!!! info  
    On standard Windows settings, Malware can hide itself if the filename is like: `Secure-File.txt.exe`

    What you see? A file named `Secure-File.txt`

    Of course the attacker can add a different icon to the file, so it looks like you open the file type extension you think.

    And if you open it, the Malware start's. See Crypto trojans.

>     Just Open the File Explorer's Settings and change it to show File Extensions by clicking on `View` > `Show`

Reviewer: ```suggestion
    Just Open the File Explorer's settings and change it to show File Extensions by clicking on `View` > `Show`
```

As you are using Windows 11, TPM is used to encrypt and decrypt the drive but it is susceptible to [cold boot attacks](https://blog.elcomsoft.com/2021/01/understanding-bitlocker-tpm-protection/). So, it is recommended to use TPM + PIN by configuring the Group Policy Editor by going to the location below and selecting things as shown in the image below it: 

```
Computer Configuration > Administrative Templates > Windows Components > Bitlocker Drive Encryption > Operating System Drives > Require Additional Authentication at startup
```

![TPM + PIN](/assets/img/windows/TPM%20%2B%20PIN%20Bitlocker.webp)

Also enable this Group Policy

```
Computer Configuration > Administrative Templates > Windows Components > Bitlocker Drive Encryption > Operating System Drives > Allow Enchanced PIN for startup

```
![Enhanced Pins](/assets/img/windows/enhanced-pins.webp)

This will allow to you use Alphanumeric characters for the Pre-boot PIN.

After enabling these policies, you need to create your PIN by opening **Windows Terminal** as an **administrator**.

You can copy/paste `manage-bde -protectors -add c: -TPMAndPIN` and it will prompt you to enter a new PIN.

> You can check if it's enabled by typing `manage-bde -status`. It will normally show in **Key Protectors** : Numerical Password (it's the recovery key) and TPM And PIN.

Reviewer: ```suggestion
You can check if it's enabled by typing `manage-bde -status`. It will normally show in **Key Protectors** : **Numerical Password** (it's the recovery key) and **TPM And PIN**.
```

---
title: System Hardening
icon: material/monitor-lock
---

## Setting up Windows after Installation

> You should select the offline account instead of a Microsoft account as shown below. With the release of Windows 11, Microsoft changed the UI to make users to use a Microsoft account by default, but the option to use an offline account is available to Education and Enterprise Edition users only. Microsoft may begin forcing users to have a [Microsoft account on Windows 11 Pro](https://www.windowslatest.com/2022/02/19/windows-11-pro-may-require-microsoft-account-but-local-option-is-here-to-stay/) soon.

Reviewer 2: ```suggestion
If you wish to limit the amount of data Microsoft obtains from your device, an offline/local account is recommended. Instructions on how to create a local Windows account are here: [How to Create a Local or Offline Account in Windows 11 or Windows 10](https://answers.microsoft.com/en-us/windows/forum/all/how-to-create-a-local-or-offline-account-in/95097c32-40c4-48c0-8f3b-3bcb67afaf7c). For security, it's recommended to set up Windows Hello on all of your accounts because it utilizes the trusted platform module (TPM) if applicable, which protects against brute-force attacks; see the documentation here: [How Windows Uses the TPM](https://docs.microsoft.com/en-us/windows/security/information-protection/tpm/how-windows-uses-the-tpm#windows-hello-for-business)
```

![user-account](/assets/img/windows/user-account.webp)

!!! note
    Microsoft is pushing users to use Microsoft accounts for other editions except Education and Enterprise after installation. 

    So, You could follow the guide by [ghacks.net](https://www.ghacks.net/2022/05/13/how-to-bypass-the-microsoft-account-requirement-during-windows-setup/) to bypass the Microsoft account requirement during setup and use Local account.

While setting up, it is recommended to use a generic names such as `user` and `host` and avoid identifying terms such as your name or operating system. This can make it more difficult for privileged `Win32` apps or attackers to discern your identity.

- [ ] Toggle off all privacy related settings as shown in the image :

![Privacy Settings](/assets/img/windows/privacy-settings.webp)

## Encrypting the Drive

> After you have installed Windows, turn on Full Disk Encryption (FDE) using BitLocker via Control Panel. 

Reviewer 2: ```suggestion
After you have installed Windows, turn on full disk encryption (FDE) using BitLocker via the Control Panel. 
```

---
title: System Hardening
icon: material/monitor-lock
---

## Setting up Windows after Installation

> If you wish to limit the amount of data Microsoft obtains from your device, an offline/local account is **recommended**. Instructions on how to create a local Windows account are here: [How to Create a Local or Offline Account in Windows 11 or Windows 10](https://answers.microsoft.com/en-us/windows/forum/all/how-to-create-a-local-or-offline-account-in/95097c32-40c4-48c0-8f3b-3bcb67afaf7c). For security, it's recommended to set up Windows Hello on all of your accounts because it utilizes the trusted platform module (TPM) if applicable, which protects against brute-force attacks; see the documentation here: [How Windows Uses the TPM](https://docs.microsoft.com/en-us/windows/security/information-protection/tpm/how-windows-uses-the-tpm#windows-hello-for-business)

Reviewer: ```suggestion
If you wish to limit the amount of data Microsoft obtains from your device, an [offline/local account](https://answers.microsoft.com/en-us/windows/forum/all/how-to-create-a-local-or-offline-account-in/95097c32-40c4-48c0-8f3b-3bcb67afaf7c) is **recommended**. https://answers.microsoft.com/en-us/windows/forum/all/how-to-create-a-local-or-offline-account-in/95097c32-40c4-48c0-8f3b-3bcb67afaf7c. For security, it's recommended to set up Windows Hello on all of your accounts because it uses the trusted platform module (TPM) if applicable, which protects against brute-force attacks; see the documentation: [How Windows Uses the TPM](https://docs.microsoft.com/en-us/windows/security/information-protection/tpm/how-windows-uses-the-tpm#windows-hello-for-business)
```

![user-account](/assets/img/windows/user-account.webp)

!!! note
    Microsoft is pushing users to use Microsoft accounts for other editions except Education and Enterprise after installation. 

    So, You could follow the guide by [ghacks.net](https://www.ghacks.net/2022/05/13/how-to-bypass-the-microsoft-account-requirement-during-windows-setup/) to bypass the Microsoft account requirement during setup and use Local account.

> While setting up, it is recommended to use a generic names such as `user` and `host` and avoid identifying terms such as your name or operating system. This can make it more difficult for privileged `Win32` apps or attackers to discern your identity.

Reviewer: ```suggestion
While setting up, it is recommended to use a generic name such as `user` and `host` and avoid identifying terms such as your name or operating system. This can make it more difficult for privileged `Win32` apps or attackers to discern your identity.
```

> - [ ] Toggle off all privacy related settings as shown in the image :

Reviewer: ```suggestion
- [ ] Toggle off all privacy related settings as shown in the image:
```

![Privacy Settings](/assets/img/windows/privacy-settings.webp)

## Encrypting the Drive

After you have installed Windows, turn on full disk encryption (FDE) using BitLocker via the Control Panel. 

!!! info "Choosing the Way to Encrypt"
    It is recommended to use only the Control Panel because if you go to encrypt via settings app, Microsoft named it as `Device Encryption` and designed it in a way that the encryption keys for BitLocker would be stored on Microsoft's server which is attached to your Microsoft account. This can be dangerous to your privacy and security as Microsoft could easily view your encrypted files, as could an attacker if they were able to gain access to Microsoft's servers or any Law Enforcement could by a Gag order. 

> Perfect way to do is go to Control Panel by searching for it in the Start Menu or from the context menu (right-click) in File Explorer and set it up for all of the drives that you have.

Reviewer: ```suggestion
The best way is to go to the Control Panel by searching for it in the Start Menu or from the context menu (right-click) in File Explorer and set it up for all of the drives that you have.
```

![Bitlocker in Control Panel](/assets/img/windows/bitlocker-control%20panel.webp)

As you are using Windows 11, TPM is used to encrypt and decrypt the drive but it is susceptible to [cold boot attacks](https://blog.elcomsoft.com/2021/01/understanding-bitlocker-tpm-protection/). So, it is recommended to use TPM + PIN.

### Setting up Pre-boot Authentication

 To setup TPM + PIN. Configure the Group Policy Editor by going to the location below and selecting things as shown in the image below it: 

```
Computer Configuration > Administrative Templates > Windows Components > Bitlocker Drive Encryption > Operating System Drives > Require Additional Authentication at startup
```

![TPM + PIN](/assets/img/windows/TPM%20%2B%20PIN%20Bitlocker.webp)

Also enable this Group Policy:

```
Computer Configuration > Administrative Templates > Windows Components > Bitlocker Drive Encryption > Operating System Drives > Allow Enchanced PIN for startup

```
![Enhanced Pins](/assets/img/windows/enhanced-pins.webp)

> This will allow to you use alphanumeric characters for the pre-boot PIN.

Reviewer: ```suggestion
This will allow you to use alphanumeric characters for the pre-boot PIN.
```

After enabling these policies, you need to create your PIN by opening a **terminal** as an **administrator**.

> You can copy/paste this command `manage-bde -protectors -add c: -TPMAndPIN` in terminal and it will prompt you to enter a new PIN.

Reviewer: ```suggestion
You can copy/paste this command `manage-bde -protectors -add c: -TPMAndPIN` in the terminal and it will prompt you to enter a new PIN.
```

> You can check if it's enabled by typing `manage-bde -status`. It will normally show in **Key Protectors** : **Numerical Password** (it's the recovery key) and **TPM And PIN**.

Reviewer: ```suggestion
You can check if it's enabled by typing `manage-bde -status`. It will normally show in **Key Protectors**: **Numerical Password** (it's the recovery key) and **TPM And PIN**.
```

??? abstract "Disabling pre-boot Authentication (Not Recommended)"

>     If you want to disable pre-boot PIN requirements if it doesn't fit in your threat model. Set the 2 policies back to **Not configured**.

Reviewer: ```suggestion
    If you want to disable pre-boot PIN requirements if it doesn't fit in your threat model. Set the two policies back to **Not configured**.
```

    The pre-boot PIN not only protects the OS drive but also other fixed drives used just for storage.

!!! tip
    To go to Computer Configuration, search **Group Policy** in the **Windows Search Bar** and press **Enter** or type `gpedit.msc` in ++win+r++ You will see it and proceed as mentioned above.

When you do this, the encryption keys of your drive are only unlocked once you enter the PIN, and the decryption happens after. If you forget or lose the PIN, you won't be able to access your drives and OS anymore, and the only way to recover is using the Recovery Key provided during the initial setup of BitLocker. **Make sure you store it in a safe place**, such as a password manager, and keep backups of your Recovery Key or even use an encrypted USB drive.

## Security Hardening

### Creation of User Account and usage

- By Default Windows gives `administrator` access to the user account. Create another `standard` user account to reduce the attack surface enormously as most vulnerabilities today come from the fact that the user is always in `administrator` mode. In addition, you shouldn't use the same password for standard and administrator account.

> - Don't use admin account for any of your personal tasks ! 

Reviewer: ```suggestion
- Don't use admin account for any of your personal tasks!
```

    
- Only use your account for you, if someone needs to use your computer **ALWAYS** create another standard account, even if it's for a one-time use, even if it's your family or someone you trust. This person can plug a malicious USB, can connect to malicious Wi-Fi network, download infectious files, etc... without you knowing about it.

- You might be afraid that the user accessing your device via another User account can access your Internal drive and access critical files violating your privacy. Refer, [Privacy page](windows/privacy/#restrict-access-to-data-drives) on How to restrict access to certain drives only or use EFS on a per-file basis. You can read more about EFS on [Privacy page]().

!!! tip
    You should ALWAYS do the quick shortcut ++win+l++ to lock your device when you are away to prevent unauthorized access.

### Securing the Bootloader

#### Enabling Secure Boot

> - Windows 11 secures it bootloader by default by using Secure boot with the usage of TPM.

Reviewer: ```suggestion
- Windows 11 secures its bootloader by default by using Secure boot with the usage of TPM.
```

> - Windows 10 on the other hand doesn't come with Secure boot enabled by default except new Devices. 

Reviewer: ```suggestion
- Windows 10, on the other hand, doesn't come with Secure boot enabled by default except for new devices. 
```

To enable Secure Boot from the PC BIOS menu. Follow this Step-by-Step Instructions by visiting this [documentation](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/disabling-secure-boot?view=windows-11#re-enable-secure-boot).

> Visit : [https://docs.microsoft.com/en-us/mem/intune/user-help/you-need-to-enable-secure-boot-windows#check-secure-boot-status](https://docs.microsoft.com/en-us/mem/intune/user-help/you-need-to-enable-secure-boot-windows#check-secure-boot-status) on how to verify if enabled after enabling secure boot.

Reviewer: ```suggestion
Visit: [https://docs.microsoft.com/en-us/mem/intune/user-help/you-need-to-enable-secure-boot-windows#check-secure-boot-status](https://docs.microsoft.com/en-us/mem/intune/user-help/you-need-to-enable-secure-boot-windows#check-secure-boot-status) on how to verify if enabled after enabling secure boot.
```

**System Guard** lets the system freely boot into untrusted code initially, but shortly after launches the system into a trusted state by taking control of all CPUs and forcing them down a well-known and measured code path. This has the benefit of allowing untrusted early UEFI code to boot the system, but then being able to securely transition into a trusted and measured state.

- [x] Enable [System Guard](https://docs.microsoft.com/en-us/windows/security/threat-protection/windows-defender-system-guard/how-hardware-based-root-of-trust-helps-protect-windows) by following the instructions of [Microsoft Docs](https://docs.microsoft.com/en-us/windows/security/threat-protection/windows-defender-system-guard/system-guard-secure-launch-and-smm-protection) to secure the boot chain.

You can also know how to check if it is enabled or not in the guide.

!!! note
    System Guard is mostly available on Windows Secured-Core PCs not on regular consumer devices. So, Before enabling it check the requirements of your Device.

### Security Improvements

> 

Reviewer: #### Address Space Layout Randomization

[ASLR](https://wonderfall.space/buffer-overflow#au-niveau-du-syst%C3%A8me-aslr) is a feature that prevents [memory corruption](https://en.wikipedia.org/wiki/Memory_corruption) like [buffer overflow](https://en.wikipedia.org/wiki/Buffer_overflow).
You can enable `ASLR mandatory` & `Bottom-up-ASLR` by going in the Defender Exploit Guard settings. 

![ASLR](/assets/img/windows/aslr.webp)

### Protection against Malware and Viruses

> - Just use the built-in Windows Defender Security to protect against threats and stick to it. Don't use anyother Antivirus or Anti-Malware software.

Reviewer: ```suggestion
- Just use the built-in Windows Defender Security to protect against threats and stick to it. Don't use any other Antivirus or Anti-Malware software [as it can weaken your security and your privacy](https://wonderfall.space/windows-hardening/#microsoft-defender-antivirus).
```

- [ ] **Enable** [Windows Defender in a Sandbox](https://www.microsoft.com/security/blog/2018/10/26/windows-defender-antivirus-can-now-run-in-a-sandbox/) by launching a **terminal** as an **administrator** and copy/paste this command ```setx /M MP_FORCE_USE_SANDBOX 1```. Restart your device and check if there's a process called **MsMpEngCP.exe** by typing `tasklist` in the terminal.
    
- [ ] **Disable** `Automatic Sample Submission` in Windows Defender will send your files as a sample for Signature Database and might leak your data. You can do it via the below Group Policy so to not prompt you again and again constantly.
```
 Computer Configuration > Administrative Templates > Windows Components > Microsoft Defender Antivirus > MAPS > Send file samples when further analysis is required to Never Send.
 ```

- [x] Enable [Controlled Folder Access](https://docs.microsoft.com/en-us/microsoft-365/security/defender-endpoint/enable-controlled-folders) in Windows defender settings. So, The Important folders you listed for protection doesn't get attacked or held hostage in case of a ransomware attack.

> !!! warning

Reviewer: ```suggestion
```

>     Enabling Controlled Folder Access will result in delayed startup of Video playback or result in someother videoplayback issues.

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

```
Computer Configuration > Administrative Templates > Windows Components > Bitlocker Drive Encryption > Operating System Drives
```
<!-- Your path is too long, if there's a shorter way to get here, use that instead -->
![TPM + PIN](/assets/img/windows/TPM%20%2B%20PIN%20Bitlocker.png)

When you do this, the encryption keys of your drive are only unlocked once you enter the PIN, and the decryption happens after. If you forget or lose the PIN, you won't be able to access your drives and OS anymore, and the only way to recover is using the Recovery Key provided during the initial setup of BitLocker. **Make sure you store it in a safe place**, such as a password manager, and keep backups of your Recovery Key.

*[TPM]: Trusted Platform Module

> *[FDE]: Full Disk Encryption

Author: I will check the Privacy Section manually and add it later.

    - **Disable** `Automatic Sample Submission` in Windows Defender settings as Windows Defender will send your files as a sample for Signature Database and might leak your data.
    - **Enable** the Group Policy `Computer Configuration > Administrative Templates > Windows Components > Microsoft Defender Antivirus > MAPS > Send file samples when further analysis is required to Never Send.`

> - [x] Enable [Microsoft Defender Application Guard](https://docs.microsoft.com/en-us/windows/security threat-protection/microsoft-defender-application-guard/md-app-guard-overview). After installing by going to "[Turn Windows Features on or off](https://docs.microsoft.com/en-us/windows/security/threat-protection/microsoft-defender-application-guard/install-md-app-guard)" you can enable it. This runs Microsoft Edge in an Isolated Hyper-V container preventing unknown Malware from damaging the system. **This is specific to Microsoft Edge only**.

Reviewer 3: I don't think you need to say `**This is specific to Microsoft Edge only**` since you've already removed the `and other **Microsoft** Apps such as MS office,etc` part. You're basically just saying the same thing twice.

Author: I understand what you mean. If the readers read the Microsoft documentation. They will know it.

    Of course the attacker can add a different icon to the file, so it looks like you open the file type extension you think.

    And if you open it, the Malware start's.

    Just Open the File Explorer's settings and change it to show File Extensions by clicking on `View` > `Show`

>     From : [beerisgood](https://github.com/beerisgood/Windows11_Hardening/blob/master/always%20display%20file%20typ%20extension)

Reviewer 3: Missed this earlier, but you can make the title of the container a link by using:
```markdown
!!! example "[Source: beerisgood](https://github.com/beerisgood/Windows11_Hardening/blob/master/always%20display%20file%20typ%20extension)"
    <enter text here>
```
or something similar.

>     From : [beerisgood](https://github.com/beerisgood/Windows11_Hardening/blob/master/always%20display%20file%20typ%20extension)

Author: Explain about the first one.

    From : [beerisgood](https://github.com/beerisgood/Windows11_Hardening/blob/master/always%20display%20file%20typ%20extension)

!!! check "Keep your Device Up-to-date"
    You should keep your Windows Device Up-to-date by enable Automatic Updating. It is recommended to do so to keep your Device free form Zero-Day Exploits.
    

>     To get know about information about updates that arrive monthly. It is recommended that you follow [Windows Latest](https://windowslatest.com).

Author: I never knew it existed. Thanks.

>     It is recommended to stick to Driver Updates given via Optional Updates as it is thoroughly vetted by Microsoft for Stability of your Device not rely on third Party apps for updating. So that, you get the latest patches for your Firmware along with Drivers to avoid any type of security vulnerabilites in the firmware too along with the OS.

Author: I am gonna add yours. It is better.
